static/app/components/core/compactSelect/control.tsx

```diff
@@ -412,24 +397,6 @@ export function Control({
     [search]
   );
 
-  /**
-   * A list of selected options across all select regions, to be used to generate the
-   * trigger label.
-   */
-  const [selectedOptions, setSelectedOptions] = useState<
-    Array<SelectOption<SelectKey> | Array<SelectOption<SelectKey>>>
-  >([]);
-  const saveSelectedOptions = useCallback<SelectContextValue['saveSelectedOptions']>(
-    (index, newSelectedOptions) => {
-      setSelectedOptions(current => [
-        ...current.slice(0, index),
-        newSelectedOptions,
-        ...current.slice(index + 1),
-      ]);
-    },
-    []
-  );
```

Comment:
no more internal state that needs to be synced just to calculate a trigger label